test/e2e_node/runner/run_e2e.go

```diff
@@ -19,100 +19,56 @@ limitations under the License.
 package main
 
 import (
-	"bufio"
 	"flag"
-	"fmt"
 	"os"
-	"os/exec"
 	"strings"
-	"sync"
-	"time"
 
-	"runtime"
+	"k8s.io/kubernetes/test/e2e_node"
 
 	"github.com/golang/glog"
-	"k8s.io/kubernetes/test/e2e_node/gcloud"
-	"path/filepath"
 )
 
-type RunFunc func(host string, port string) ([]byte, error)
-
-type Result struct {
-	host   string
-	output []byte
-	err    error
-}
-
 const gray = "\033[1;30m"
 const blue = "\033[0;34m"
 const noColour = "\033[0m"
 
-var u = sync.WaitGroup{}
-var zone = flag.String("zone", "", "gce zone the hosts live in")
 var hosts = flag.String("hosts", "", "hosts to test")
-var wait = flag.Bool("wait", false, "if true, wait for input before running tests")
-var kubeOutputRelPath = flag.String("k8s-build-output", "_output/local/bin/linux/amd64", "Where k8s binary files are written")
-
-var kubeRoot = ""
-
-const buildScriptRelPath = "hack/build-go.sh"
-const ginkgoTestRelPath = "test/e2e_node"
-const healthyTimeoutDuration = time.Minute * 3
 
 func main() {
 	flag.Parse()
 	if *hosts == "" {
 		glog.Fatalf("Must specific --hosts flag")
 	}
-
-	// Figure out the kube root
-	_, path, _, _ := runtime.Caller(0)
-	kubeRoot, _ = filepath.Split(path)
-	kubeRoot = strings.Split(kubeRoot, "/test/e2e_node")[0]
-
-	// Build the go code
-	out, err := exec.Command(filepath.Join(kubeRoot, buildScriptRelPath)).CombinedOutput()
-	if err != nil {
-		glog.Fatalf("Failed to build go packages %s: %v", out, err)
-	}
-
-	// Copy kubelet to each host and run test
-	if *wait {
-		u.Add(1)
-	}
+	archive := e2e_node.CreateTestArchive()
+	defer os.Remove(archive)
 
 	results := make(chan *TestResult)
 	hs := strings.Split(*hosts, ",")
 	for _, h := range hs {
-		go func(host string) { results <- runTests(host) }(h)
-	}
-
-	// Maybe wait for user input before running tests
-	if *wait {
-		WaitForUser()
+		glog.Infof("Starting tests on host %s.", h)
+		go func(host string) {
+			output, err := e2e_node.RunRemote(archive, host)
+			results <- &TestResult{
+				output: output,
+				err:    err,
+				host:   host,
+			}
+		}(h)
 	}
 
 	// Wait for all tests to complete and emit the results
 	errCount := 0
 	for i := 0; i < len(hs); i++ {
 		tr := <-results
-		host := tr.fullhost
+		host := tr.host
+		glog.Infof("%s================================================================%s", blue, noColour)
```

Comment:
(optional) Most unix utilities only add color if the output is not piped, consider doing the same (or at least providing a flag).
```go
stat, _ := os.Stdout.Stat()
useColor := (stat.Mode() & os.ModeCharDevice) != 0
```